embossc

```diff
@@ -96,10 +102,15 @@ def main(argv):
   if back_end_status.returncode != 0:
     return back_end_status.returncode
 
-  output_file = os.path.join(flags.output_path[0], flags.input_file[0]) + ".h"
-  os.makedirs(os.path.dirname(output_file), exist_ok=True)
+  if flags.output_file:
+    output_file = flags.output_file[0]
+  else:
+    output_file = flags.input_file[0] + ".h"
+
+  output_filepath = os.path.join(flags.output_path[0], output_file)
+  os.makedirs(os.path.basename(os.path.dirname(output_filepath)), exist_ok=True)
```

Comment:
Why did you add the `basename` call?  That's a breaking behavioral change.

Reply:
Ah, that is a merge mistake on my part. Removed.